src/content/it/fundamentals/performance/critical-rendering-path/render-blocking-css.md

```diff
@@ -1,56 +1,78 @@
+
+
+L'esempio di cui sopra mostra il sito web di NYTimes con e senza CSS, dimostra
+il motivo per cui il rendering è bloccato finché CSS è disponibile, senza CSS la
+pagina è in realtà inutilizzabile. In effetti, l'esperienza alla destra viene
+spesso definita come "Flash of Unstyled Content" (FOUC). Di conseguenza, il
+browser bloccherà il rendering finché non disporrà sia di DOM che di CSSOM.
+
+> **_CSS è una risorsa che blocca il rendering, trascriverla quanto prima
+> possibile per ottimizzare il tempo del primo rendering._**
+
+Tuttavia, se invece disponessimo di alcuni stili CSS che sono utilizzati solo in
+determinate circostanze, ad esempio quando la pagina viene stampata o proiettata
+su un grande monitor? Sarebbe bello se non avessimo da bloccare il rendering di
+queste risorse.
+
+I "media types" e le "media queries" CSS ci permettono di affrontare questi casi di utilizzo:
 
 
     <link href="style.css" rel="stylesheet">
     <link href="print.css" rel="stylesheet" media="print">
     <link href="other.css" rel="stylesheet" media="(min-width: 40em)">
     
 
-Una [query supporti](/web/fundamentals/design-and-ux/responsive/#use-media-queries) è costituita da un tipo di supporto e zero o più espressioni che controllano le condizioni di quelle particolari funzioni del supporto. Ad esempio, la nostra prima dichiarazione del foglio di stile non fornisce alcun tipo di tipo di supporto o di query supporti, dunque sarà applicata in tutti i casi, questo per dire che bloccherà sempre il rendering. D'altro canto, il secondo foglio di stile si applicherà solamente quando il contenuto viene stampato: magari vuoi riorganizzare il layout, cambiare i caratteri e così via e quindi questo foglio di stile non deve bloccare il rendering della pagina al primo caricamento. Infine, l'ultima dichiarazione del foglio di sitle fornisce una 'query supporti' che viene eseguita dal browser: se le condizioni corrispondono, allora il browser bloccherà il rendering finché il foglio di stile non viene scaricato ed elaborato.
+Una [media query](/web/fundamentals/design-and-ux/responsive/#use-media-queries) è costituita da un tipo di supporto e zero o più espressioni che controllano le condizioni di quelle particolari funzioni del supporto. Ad esempio, la nostra prima dichiarazione del foglio di stile non fornisce alcun tipo di tipo di supporto o di query supporti, dunque sarà applicata in tutti i casi, questo per dire che bloccherà sempre il rendering. D'altro canto, il secondo foglio di stile si applicherà solamente quando il contenuto viene stampato: magari vuoi riorganizzare il layout, cambiare i caratteri e così via e quindi questo foglio di stile non deve bloccare il rendering della pagina al primo caricamento. Infine, l'ultima dichiarazione del foglio di sitle fornisce una 'query supporti' che viene eseguita dal browser: se le condizioni corrispondono, allora il browser bloccherà il rendering finché il foglio di stile non viene scaricato ed elaborato.
```

Comment:
Una [media query](/web/fundamentals/design-and-ux/responsive/#use-media-queries) è costituita da un media type e zero o più espressioni che controllano le condizioni di quelle particolari funzioni media. Ad esempio, la nostra prima dichiarazione del foglio di stile non fornisce alcun media type o media query, dunque sarà applicata in tutti i casi, questo per dire che bloccherà sempre il rendering. D'altro canto, la dichiarazione del secondo foglio di stile si applicherà solamente quando il contenuto viene stampato: magari vuoi riorganizzare il layout, cambiare i caratteri e così via e quindi questo foglio di stile non deve bloccare il rendering della pagina al primo caricamento. Infine, l'ultima dichiarazione del foglio di sitle fornisce una "media query" che viene eseguita dal browser: se le condizioni corrispondono, allora il browser bloccherà il rendering finché il foglio di stile non viene scaricato ed elaborato.